### ISSUE TITLE
Filecrypt puzzlecaptcha (cutcaptcha)

### ISSUE THREAD

**Issue by j4m** [pyload/pyload#3295]
Is it possible to solve puzzle captchas with pyload?

<img width="410" alt="cutcaptcha" src="https://user-images.githubusercontent.com/5821759/50378360-d009a280-0630-11e9-8dae-f6fe92a99a28.png">

Example: 
https://filecrypt.cc/Container/7403B8F996.html

**Comment by Yumekui** [comment]
quickly looked at the javascript code. pyload would need to generate 'cap_token' with the pixel coordinates. the javascript code that generates it is sadly obfuscated.

**Comment by GammaC0de** [comment]
@Yumekui I already have a theoretically working plugin,
The issue is that the javascript detects that it runs inside a iframe and not runs when does..

**Comment by Quicksteve** [comment]
@GammaC0de do you think you could share what you are doing to solve the captcha?

**Comment by GammaC0de** [comment]
Run this proof of concept html file inside your browser:
[cutcaptcha.zip](https://github.com/pyload/pyload/files/3269958/cutcaptcha.zip)

The issue is that this html will not run inside an iframe.
If you know how to fool it to run then we can make progress..

**Comment by Quicksteve** [comment]
It appears there's a misunderstanding, I thought you created something to automatically move the pieces to the correct location. I might get a chance to play around with the ball of java script to understand it and see what it does to detect iframe.

**Comment by snickers2k** [comment]
is there any progress made on this? i need aswell a way of automatic solving these via 9kw or any other captcha solver. thank you

**Comment by GammaC0de** [comment]
Does 9kw support cutcaptcha?

I had send a query long ago (2018) to 9kw via https://www.9kw.eu/ticket.html and never got an answer..

**Comment by thenewcoderintheworld** [comment]
> is there any progress made on this? i need aswell a way of automatic solving these via 9kw or any other captcha solver. thank you
>
Progress was not in Pyload but in JDownloader. Cutcaptcha with JD was possible but then broken again (changes in the cutcaptcha code). See this ticket https://svn.jdownloader.org/issues/87583 (you see only all details with jd redmine account) and forum threads https://board.jdownloader.org/showthread.php?t=81489 + https://board.jdownloader.org/showthread.php?t=82061 + https://board.jdownloader.org/showthread.php?t=82096
But jd devs are out of time because temp. fix (localfile) and long fix are thinkable (js changes in real time, cutcaptcha has blocked the browser solver  under localhost).

> Does 9kw support cutcaptcha?
>
Yes and no. 9kw can puzzle captchas as multiple images like keycaptcha but cutcaptcha is planned as interactive captcha like recaptcha in the future.

> I had send a query long ago (2018) to 9kw via ticket and never got an answer..
>
Re-submit it if you never got an answer and my answer is not enough in this case.

**Comment by StefftheEmperor2** [comment]
I think this needs the captha- interactive script changed from temperscript to native extensions, like @GammaC0de suggested. There is a request api to modify the body of responses. The first cutcaptcha script could be spoofed by mocking the top location. But it is loading another iframe, which the pyload window can not access.

**Comment by snickers2k** [comment]
> Does 9kw support cutcaptcha?
> 
> I had send a query long ago (2018) to 9kw via https://www.9kw.eu/ticket.html and never got an answer..

in one post one of the dev's of 9kw said they do handle cutcaptcha as a normal puzzlecaptcha.

**Comment by GammaC0de** [comment]
@thenewcoderintheworld Do you know how to contact 9kw team?
I did not get any response for my CutCaptcha support question and now I even opened a new ticket regarding HCaptcha support an I'm facing the same experience: total disregard 😢

**Comment by thenewcoderintheworld** [comment]
> @thenewcoderintheworld Do you know how to contact 9kw team?
Usually email and ticket system. Direct contact (e.g. messengers like jabber, telegram and more) for developers thinkable as request with reason (e.g. for faster developer support).

> I did not get any response for my CutCaptcha support question and now I even opened a new ticket regarding HCaptcha support an I'm facing the same experience: total disregard

Ticket id in this case? 
9kw.eu API doc is already for HCaptcha online with all details and it's already active in JD2 (open source).

**Comment by GammaC0de** [comment]
@thenewcoderintheworld I finally got an answer for my ticket (I guess it's because of your help, so thanks :-) ).

I have looked at the JD2 code and added HCaptcha support but there is an issue:
I cannot get a solution for my HCaptcha solving requests (I could get an answer only once 😢 ),
I can see that [9kw client](https://www.9kw.eu/software_9kwclient.html) does not support HCaptcha solving, so how can 9kw can support HCaptcha without support for it in 9kw client?

P.S. Can you do me a favior andsend me the how to contact the 9kw devs?  my email is nitzo2001 at yahoo dot com. It can really help in the future because I am not a member ib the JS forums, thanks.
